server: tenantID field unnused in api_v2 code

Found in CI linter: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/7502335?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
------- Stdout: -------
=== RUN   TestLint/TestStaticCheck
    lint_test.go:1751:
        pkg/server/api_v2.go:103:2: field tenantID is unused (U1000)
    --- FAIL: TestLint/TestStaticCheck (955.08s)
```



Jira issue: CRDB-21447

--- a/pkg/server/api_v2.go
+++ b/pkg/server/api_v2.go
@@ -44,7 +44,6 @@ import (
 	"strconv"
 
 	"github.com/cockroachdb/cockroach/pkg/kv"
-	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
@@ -82,7 +81,6 @@ type apiV2ServerOpts struct {
 	admin            *adminServer
 	status           *statusServer
 	promRuleExporter *metric.PrometheusRuleExporter
-	tenantID         roachpb.TenantID
 	sqlServer        *SQLServer
 	db               *kv.DB
 }
@@ -100,7 +98,6 @@ type apiV2Server struct {
 	status           *statusServer
 	promRuleExporter *metric.PrometheusRuleExporter
 	mux              *mux.Router
-	tenantID         roachpb.TenantID
 	sqlServer        *SQLServer
 	db               *kv.DB
 }

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -1663,7 +1663,6 @@ func (s *Server) PreStart(ctx context.Context) error {
 			admin:            s.admin,
 			status:           s.status,
 			promRuleExporter: s.promRuleExporter,
-			tenantID:         roachpb.SystemTenantID,
 			sqlServer:        s.sqlServer,
 			db:               s.db,
 		}), /* apiServer */

--- a/pkg/server/tenant.go
+++ b/pkg/server/tenant.go
@@ -572,7 +572,6 @@ func (s *SQLServerWrapper) PreStart(ctx context.Context) error {
 		s.debug,           /* handleDebugUnauthenticated */
 		newAPIV2Server(workersCtx, &apiV2ServerOpts{
 			sqlServer: s.sqlServer,
-			tenantID:  s.sqlCfg.TenantID,
 			db:        s.db,
 		}), /* apiServer */
 	); err != nil {

